teps/0137-cloudevents-controller.md

```diff
@@ -0,0 +1,527 @@
+
+- Tekton v2 events and CDEvents could be implemented without using a dedicated
+  controller for events
+- Tekton v1 events could be moved to a dedicated controller for events with no
+  support for CDEvents
+- Only core CDEvents events could be supported (`TaskRun` and `PipelineRun`
+  events)
+
+
+## Implementation Plan
+
+The most pressing feature for this TEP is the support for Tekton `v1` API
+in events, which is done through the introduction of `v2` events. Because
+of that, the implementation plan puts that towards the beginning of the
+implementation plan:
+
+- Create a dedicated config map for events configuration. Deprecate the current
+  settings in the `defaults` config map. Log warning in case of inconsistent
+  configuration, and let the config map take over when used
+- Relocate the existing CloudEvents logic (v1 events) to the events controller
+- Document operational options for the events controller (disable, scale, tune
+  RBAC, maintenance)
+- Introduce a config flag for the event format to be produced. Initial value
+  is `v1` only. The controller may send multiple events if multiple values are
+  selected, although only one is available for now
+- Restructure the cache module to exclude the event type from the key.
+  Update the new cache content based filtering logic accordingly and introduce
+  the `v1` queued event for `CustomRuns`
+- Move `TaskRun` events (for `v1`) to the external controller
+- Move `PipelineRun` events (for `v1`) to the external controller and cleanup
+  any logic related to the main controller sending CloudEvents
+- Introduce support for `v2` events. Add the `v2` value to the list of valid
+  configurations and document the `v2` format. Deprecate the `v1` format.
+- Introduce support for core CDEvents (`PipelineRun` and `TaskRun` events) with
+  the corresponding configuration flag and documentation
+- Introduce support for other CDEvents to let task author emit reacher events
+  during pipeline execution
+
+
+### Test Plan
+
+- Shared testing package for controller unit tests
+- Unit tests, reconcile and controller tests for each controller
+
+### Infrastructure Needed
+
+<!--
+(optional)
+
+Use this section if you need things from the project or working group.
+Examples include a new subproject, repos requested, GitHub details.
+Listing these here allows a working group to get the process for these
+resources started right away.
+-->
+
+### Upgrade and Migration Strategy
+
+<!--
+(optional)
+
+Use this section to detail whether this feature needs an upgrade or
+migration strategy. This is especially useful when we modify a
+behavior or add a feature that may replace and deprecate a current one.
+-->
+
+### Implementation Pull Requests
+
+- [New config map](https://github.com/tektoncd/community/pull/1028)
+- [Rename customrun package](https://github.com/tektoncd/pipeline/pull/6884)
+- [Restructure customrun event controller](https://github.com/tektoncd/pipeline/pull/6889)
+
+## References
```

Comment:
NIT: results in an empty list - https://github.com/tektoncd/community/blob/2a50cdbb4c4ea7f83a9e502d200aa84858b265f5/teps/0137-cloudevents-controller.md#references

Reply:
Oh, nice catch, thank you